pkg/frontend/admin_openshiftcluster_etcdcertificaterenew.go

```diff
@@ -0,0 +1,431 @@
+	case err != nil:
+		return err
+	}
+
+	k, err := f.kubeActionsFactory(log, f.env, doc.OpenShiftCluster)
+	if err != nil {
+		return err
+	}
+	e := &etcdrenew{
+		log:           log,
+		k:             k,
+		secretNames:   nil,
+		mode:          "",
+		backupSecrets: make(map[string][]byte),
+		lastRevision:  0,
+	}
+
+	if err = e.validateClusterVersion(ctx); err != nil {
+		return err
+	}
+
+	// Fetch secretNames using nodeNames
+	masterNodeNames, err := fetchNodeNames(ctx, k, log)
+	if err != nil {
+		return err
+	}
+	e.log.Infof("Nodes found: %s", masterNodeNames)
+	if len(masterNodeNames) != 3 {
+		return api.NewCloudError(http.StatusForbidden, api.CloudErrorCodeForbidden, "", "The cluster doesn't have 3 master nodes, quitting.")
+	}
+	for _, nodeName := range masterNodeNames {
+		for _, prefix := range []string{"etcd-peer-", "etcd-serving-", "etcd-serving-metrics-"} {
+			e.secretNames = append(e.secretNames, prefix+nodeName)
+		}
+	}
+
+	// validate etcd and certificates, backup and delete secrets
+	if err = e.validateEtcdAndBackupDeleteSecretOnFlagSet(ctx, true); err != nil {
+		return err
+	}
+
+	// Once secrets are deleted, the operator recreates the secrets and a new etcd revision is applied
+	// On failure, proceed for recovery by applying the backupsecrets on the cluster again
+	// On success, verify the etcd state, certificates
+	err = e.isEtcDRootCertRenewed(ctx)
+	if err != nil {
+		e.log.Infoln("Attempting to recover from backup, and wait for new revision to be applied after recovery")
+		return e.revertChanges(ctx)
+	}
+
+	e.log.Infoln("Etcd certificates are renewed and new revision is applied, verifying.")
+	e.mode = "renewed"
+	return e.validateEtcdAndBackupDeleteSecretOnFlagSet(ctx, false)
+}
+
+func (e *etcdrenew) validateClusterVersion(ctx context.Context) error {
+	e.log.Infoln("validating cluster version now")
+	rawCV, err := e.k.KubeGet(ctx, "ClusterVersion.config.openshift.io", "", "version")
+	if err != nil {
+		return err
+	}
+	cv := &configv1.ClusterVersion{}
+	err = codec.NewDecoderBytes(rawCV, &codec.JsonHandle{}).Decode(cv)
+	if err != nil {
+		return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", fmt.Sprintf("failed to decode clusterversion, %s", err.Error()))
+	}
+	clusterVersion, err := version.GetClusterVersion(cv)
+	if err != nil {
+		return err
+	}
+	// ETCD ceritificates are autorotated by the operator when close to expiry for cluster running 4.9+
+	if !clusterVersion.Lt(version.NewVersion(4, 9)) {
+		return api.NewCloudError(http.StatusForbidden, api.CloudErrorCodeForbidden, "", "etcd certificate renewal is not needed for cluster running version 4.9+")
+	}
+	e.log.Infof("validated: cluster version is %s", clusterVersion)
+
+	return nil
+}
+
+func fetchNodeNames(ctx context.Context, k adminactions.KubeActions, log *logrus.Entry) ([]string, error) {
```

Comment:
Instead of having this separate function to check master node names, what's to stop us from just constructing the expected master node names like we did with the secret names? For example, the three master node names will always be clustername+infraID+master-*, unless there has been an unsupported change or master replacement. For example:
```
oc get nodes
NAME                                           STATUS   ROLES                  AGE   VERSION
steven-wi-cluster-pzqvm-master-0               Ready    control-plane,master   35d   v1.27.3+e123787
steven-wi-cluster-pzqvm-master-1               Ready    control-plane,master   35d   v1.27.3+e123787
steven-wi-cluster-pzqvm-master-2               Ready    control-plane,master   35d   v1.27.3+e123787
```

Reply:
Made the changes, thank you.